parts/openshift/unstable/openshiftnodescript.sh

```diff
@@ -13,7 +13,12 @@ echo "BOOTSTRAP_CONFIG_NAME=node-config-infra" >>/etc/sysconfig/${SERVICE_TYPE}-
 echo "BOOTSTRAP_CONFIG_NAME=node-config-compute" >>/etc/sysconfig/${SERVICE_TYPE}-node
 {{end}}
 
-rm -rf /etc/etcd/* /etc/origin/master/* /etc/origin/node/*
+
+rm -rf /etc/etcd/* /etc/origin/master/*
+
+sed -i 's#^CONFIG_FILE=.*$#CONFIG_FILE=/etc/origin/node/node-config.yaml#' /etc/sysconfig/${SERVICE_TYPE}-node
+
+cp /etc/origin/node/bootstrap-node-config.yaml /etc/origin/node/node-config.yaml 
```

Comment:
some questions about these lines - perhaps some code documentation needed

- what is put into /etc/origin/node that we need to keep on the node? (why no longer rm -rf /etc/origin/node)?
- can the change to /etc/sysconfig/${SERVICE_TYPE}-node be baked into the image (or put a # TODO here)
- what is the purpose of the cp ?

Reply:
> what is put into /etc/origin/node that we need to keep on the node? (why no longer rm -rf /etc/origin/node)?

The installer lays down a few files inside of /etc/origin/node like bootstrap-node-config.yaml which is expected to be there to start the node in order for it to perform the bootstrap process.  If it is not there I don't believe the bootstrap will work.

> can the change to /etc/sysconfig/${SERVICE_TYPE}-node be baked into the image (or put a # TODO here)

The installer creates this: `CONFIG_FILE=/etc/origin/node/bootstrap-node-config.yaml`.  This allows the bootstrap to occur on the first boot.  The issue with this is that the openshift-sdn daemonset will not start as it is looking for a valid node-config.yaml which does not exist.

> what is the purpose of the cp ?

The purpose of the `cp` is to copy the bootstrap-node-config.yaml into the place of the node-config.yaml, `sed` the sysconfig file to tell the node process to look at the correct config file for openshift-sdn, and then once the node is state ready the openshift-node sync daemonset will sync the correct node-config.yaml to the node and use it.

I can update this with some documentation but we should also try to a few iterations of seeing if the node syncing pod will update the node-config.yaml.  I tried a few different permutations but landed on this one as it was successful.  I figured we could get it working and then refine.

Reply:
> The installer creates this: CONFIG_FILE=/etc/origin/node/bootstrap-node-config.yaml. 

Does that mean that we should stop generating it in acs-engine? Probably yes?

Reply:
I think we still need to generate it for the dynamic node-config.yaml process.